packages/expo-modules-autolinking/src/autolinking/findModules.ts

```diff
@@ -32,10 +32,18 @@ export async function findModulesAsync(providedOptions: SearchOptions): Promise<
     const packageConfigPaths = await findPackagesConfigPathsAsync(searchPath);
 
     for (const packageConfigPath of packageConfigPaths) {
-      const packagePath = await fs.realpath(path.join(searchPath, path.dirname(packageConfigPath)));
-      const expoModuleConfig = requireAndResolveExpoModuleConfig(
-        path.join(packagePath, path.basename(packageConfigPath))
-      );
+      const packageDirParts = path.dirname(packageConfigPath).split(path.sep);
+      const isExpoAdapter =
```

Comment:
Does it need to be different actually? Doesn't it complicate the logic too much?

Reply:
> Does it need to be different actually? Doesn't it complicate the logic too much?

since the logic is slightly different, we used to expect `package.json` and `expo-module.config.json` in the same directory. however, in expo adapter mode, it would be `expo/expo-module.config.json`, so introduced a complicate logic here 😅  i am open to this design, please recommend whether there's any better solution.

Reply:
Would it be possible to have `expo-module.config.json` at the root dir and specify the paths to podspecs or gradle files in `expo` folder? Making this configurable may also be useful I guess 🤔

Reply:
sure, we already had this feature for podspecs. however, the goal of my design is trying to put everything in `expo/` folder when we introduce integration to 3rd party libraries. do you think whether this to be valuable?

Reply:
Such separate folder sounds reasonable, but this is what complicates the most. I would rather keep the module config at the root folder (less glob patterns to look for) and get rid of detecting whether it's an expo adapter or not. Is that possible?

Wouldn't such a config 👇 do the similar job?
```json
{
  "ios": {
    "podspecPath": "expo/RNThirdPartyExpoAdapter.podspec"
  }
}
```

Reply:
it's fair enough to me. i'll update the pr today.

Reply:
@tsapeta it's done with this
```json
{
  "android": {
    "gradlePath": "expo/android/build.gradle"
  }
}
```
i've also verified reanimated integration with the patch attached in this pr's description. please help to review again when you get a chance.